tests/test_schema.py

```diff
@@ -2042,6 +2034,8 @@ class Model(BaseModel):
 @pytest.mark.xfail(reason='working on V2')
 def test_model_with_schema_extra():
     class Model(BaseModel):
+        # TODO: Do we want to keep schema_extra as a field in ConfigDict?
+        #   If not, should I delete this test?
         model_config = ConfigDict(schema_extra={'examples': [{'a': 'Foo'}]})
```

Comment:
> To be honest, I don't remember this model_config, I guess this is equivalent to the previous inner class Config but with better typing support, etc. right?

Not sure if the typing support is better (😅), but yes, it is replacing the inner class Config.

There are new ways to update the schema, but I think they (at least as of the writing of this comment) involve a bit more ceremony than this (overriding methods, creating types, etc.).

In general I think I'm in favor of reducing the amount of stuff in model_config where appropriate, and this seems like a good opportunity; I think the main change I'll make is changing the JsonSchemaExtra dataclass I have added to just be a TypedDict. While I liked having the methods, I think it will be fine to move them elsewhere and I think using a TypedDict will make it easier for users to pick up (and with less boilerplate).

Reply:
I think we should just support `__pydantic_modify_json_schema__` (or whatever it end su being called) on a model, then anything else should only be supported for backwards compatibility.